src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -297,5 +297,5 @@ def process(
             )
             return
 
-        diff = self.deploy()
+        diff = self.deploy(True)
```

Comment:
this should be `self.deploy()`, and it should do the right thing by default. prune should default to None, and be resolved to True/False automatically depending on whether files were passed in. Much more intuitive IMHO because the knowledge of special-casing deploy with no args is in a single location.

Reply:
This is the run processor, we don't have a `--prune` flag for `snow app run`